### PR TITLE
[Terrarium - Providers]: Implementation of WebUI & REST microservice for providers support

### PR DESCRIPTION
**Pull request by anjanikshree12**
This PR includes changes:
* Implemented web UI for providers (for listing providers, for listing a specific provider's metadata).
* Implemented REST microservice for providers' data handling.
* Updated browse service to initialize the REST server defined above.

**Note**: there are some temporary changes in files due to data source being the json file for now, they will get removed eventually when we switch the datasource.

Homepage:
terrarium mainpage
Provider's page:
image
image
Provider's metadata:
metadata-handlerFunc1
image

### CODE REVIEW

File: internal/restapi/providers/v1/handler.go [hunk 1]
```diff
@@ -99,13 +99,12 @@ func (h *providersV1HttpService) downloadProviderHandler() http.Handler {
 		providerVersion, providerOS, providerArch := GetProviderInputsFromRequest(r)
 		span.SetAttributes(
 			attribute.String("module.name", providerName),
```
Comment (odelanno) [inline internal/restapi/providers/v1/handler.go:101]: Why is module name a span attribute here? 
It does not serve a module, please remove the attribute.

File: cmd/browse.go [hunk 2]
```diff
@@ -24,8 +25,13 @@ func init() {
...
 
+	vm, err := providerServices.NewJSONFileProviderVersionManager()
+	if err != nil {
+		panic(err)
+	}
+
 	restAPIServer := browse.New(registrar.NewRegistrarGrpcClient(registrar.RegistrarServiceEndpoint),
 		version_manager.NewVersionManagerGrpcClient(version_manager.VersionManagerEndpoint),
-		release.NewBrowseGrpcClient(release.ReleaseServiceEndpoint))
+		release.NewBrowseGrpcClient(release.ReleaseServiceEndpoint), vm)
```
Comment (odelanno) [inline cmd/browse.go:35]: could you use a better name for vm?

**Review by odelanno — CHANGES_REQUESTED** [review]

File: cmd/browse.go (hunk 2 above)
Comment (anjanikshree12) [reply cmd/browse.go]: Yes, updated it with version_manager_svc

**Review by odelanno — APPROVED** [review]

File: internal/restapi/providers/v1/handler.go (hunk 1 above)
Comment (anjanikshree12) [reply internal/restapi/providers/v1/handler.go]: missed to update the attribute name, it was additional and was not required, removed it.

File: cmd/allInOne.go [hunk 3]
```diff
@@ -13,6 +13,7 @@ import (
 	storage2 "github.com/terrariumcloud/terrarium/internal/module/services/storage"
 	"github.com/terrariumcloud/terrarium/internal/module/services/tag_manager"
 	"github.com/terrariumcloud/terrarium/internal/module/services/version_manager"
+	services3 "github.com/terrariumcloud/terrarium/internal/provider/services"
```
[the remark this answers is not in the document]
Comment (anjanikshree12) [reply cmd/allInOne.go]: Updated with suggested changes!

**Pull request merged**